### PR TITLE
[New] MsiExec Service Child Process With Network Connection

### PR DESCRIPTION
**Pull request by Samirbous**
converted an ER diag rule to SIEM rule (compatible with 3d party EDRs) as it matches on a good number of MSI related FNs.

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: New - Guidelines

These guidelines serve as a reminder set of considerations when proposing a new rule.

### Documentation and Context

- [ ] Detailed description of the rule.
- [ ] List any new fields required in ECS/data sources.
- [ ] Link related issues or PRs.
- [ ] Include references.

### Rule Metadata Checks

- [ ] `creation_date` matches the date of creation PR initially merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive, considering performance for diverse environments. Non ecs fields should be added to `non-ecs-schema.json` if not available in an integration.
- [ ] `min_stack_comments` and `min_stack_version` should be included if the rule is only compatible starting from a specific stack version.
- [ ] `index` pattern should be neither too specific nor too vague, ensuring it accurately matches the relevant data stream (e.g., use logs-endpoint.process-* for process data).
- [ ] `integration` should align with the `index`. If the integration is newly introduced, ensure the manifest, schemas, and `new_rule.yaml` template are updated.
- [ ] `setup` should include the necessary steps to configure the integration.
- [ ] `note` should include any additional information (e.g. Triage and analysis investigation guides, timeline templates).
- [ ] `tags` should be relevant to the threat and align/added to the `EXPECTED_RULE_TAGS` in the definitions.py file.
- [ ] `threat`, `techniques`, and `subtechniques` should map to ATT&CK always if possible.

#### New BBR Rules
- [ ] `building_block_type` should be included if the rule is a building block and the rule should be located in the `rules_building_block` folder.
- [ ] `bypass_bbr_timing` should be included if adding custom lookback timing to the rule.

### Testing and Validation

- [ ] Provide evidence of testing and detecting the expected threat.
- [ ] Check for existence of coverage to prevent duplication.

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/windows/defense_evasion_msiexec_child_proc_netcon.toml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+updated_date = "2024/09/09"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of an MsiExec service child process followed by network or dns lookup activity. Adversaries may
+abuse Windows Installers for initial access and delivery of malware.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.process-*", "logs-endpoint.events.network-*", "logs-windows.sysmon_operational-*", "endgame-*", "logs-sentinel_one_cloud_funnel.*", "logs-m365_defender.event-*"]
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_msiexec_child_proc_netcon.toml:14]:
```suggestion
index = ["logs-endpoint.events.process-*", "logs-endpoint.events.network-*", "logs-windows.sysmon_operational-*", "logs-sentinel_one_cloud_funnel.*"]
```
These don't have entity_id

File: rules/windows/defense_evasion_msiexec_child_proc_netcon.toml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+[metadata]
+creation_date = "2024/09/09"
+integration = ["endpoint", "windows", "sentinel_one_cloud_funnel", "m365_defender"]
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_msiexec_child_proc_netcon.toml:3]:
```suggestion
integration = ["endpoint", "windows", "sentinel_one_cloud_funnel"]
```

File: rules/windows/defense_evasion_msiexec_child_proc_netcon.toml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+risk_score = 47
+rule_id = "65432f4a-e716-4cc1-ab11-931c4966da2d"
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Defense Evasion",
+    "Data Source: Elastic Endgame",
+    "Data Source: Elastic Defend",
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_msiexec_child_proc_netcon.toml:27]:
```suggestion
    "Data Source: Elastic Defend",
```

File: rules/windows/defense_evasion_msiexec_child_proc_netcon.toml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    "Data Source: Sysmon",
+    "Data Source: SentinelOne",
+    "Data Source: Microsoft Defender for Endpoint"
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_msiexec_child_proc_netcon.toml:30]:
```suggestion
    "Data Source: SentinelOne",
```

**Review by w0rk3r — APPROVED** [review]

**Pull request merged**